### ISSUE TITLE
Contract level metadata

### ISSUE THREAD

**Issue by julien51** [unlock-protocol/unlock#13089]
It looks like some tools (Rainbow...) uses/requires the use of a _contract_ level metadata: https://docs.opensea.io/docs/contract-level-metadata

Since we already have the Base token URI, this is probably just an easy addition? 

Wdyt @clemsos

**Comment by clemsos** [comment]
yes it seems like an easy addition to add, with default url being the base tokenURI and the lock address. Just need to make sure we return the correct json in locksmith first?

**Comment by julien51** [comment]
> 
yes it seems like an easy addition to add, with default url being the base tokenURI and the lock address. Just need to make sure we return the correct json in locksmith first?


good point! I will work on that. It should be pretty trivial!

**Comment by julien51** [comment]
ok so this creates a challenge when people are using IPFS to host the NFT metadata, so let's hold off until we have a good way to handle this.

**Comment by juanlysander** [comment]
I recently utilized the `setLockMetadata` function in Solidity to assign an IPFS address to a lock. While the function call executes correctly, I noticed that there currently isn't a direct method to retrieve the lock metadata that has been set.

To enhance functionality, I propose the implementation of a corresponding getter function for setLockMetadata in the PublicLock contract. This addition would allow users to easily access and verify the metadata associated with a particular lock.

**Comment by juanlysander** [comment]
```solidity
// smart-contracts/contracts/mixins/MixinLockMetadata.sol

/// An abbreviated name for NFTs in this contract. Defaults to "KEY" but is settable by lock owner
  string private lockSymbol;

  // the base Token URI for this Lock. If not set by lock owner, the global URI stored in Unlock is used.
  string private baseTokenURI;
  ```
  should the lockSymbol and baseTokenURI turned into public?
  or maybe add
  ```solidity
  function getLockSymbol() public view returns (string memory) {
        return lockSymbol;
    }

    function getBaseTokenURI() public view returns (string memory) {
        return baseTokenURI;
    }
    ```
    how about it? @julien51 @clemsos

**Comment by clemsos** [comment]
> should the lockSymbol and baseTokenURI turned into public?

hey @juanlysander, thanks for the queston ! 

There is a `symbol()` function that returns the symbol already.  As @julien51 mentioned, a `baseTokenURI()` will not work as root address for IPFS, even though the current `tokenURI()` works fine.

What was your issue exactly while using IPFS? Could you describe it?
